### PR TITLE
Fix NEP-141 Compliance

### PR DESCRIPTION
**Pull request by mfornet**
Return number as json (string) in ft_on_transfer

### CODE REVIEW

**Review by sept-en — APPROVED** [review]
Good catch! LGTM!

Please fix tests and let's merge this ASAP.

File: src/fungible_token.rs [hunk 1]
```diff
@@ -245,12 +246,10 @@ impl FungibleToken {
...
             PromiseResult::Successful(value) => {
-                if let Ok(unused_amount) = String::from_utf8(value) {
-                    let unused_amount = if let Ok(v) = unused_amount.parse::<u128>() {
-                        v
-                    } else {
-                        amount
-                    };
+                if let Some(unused_amount) = parse_json(value.as_slice())
+                    .map(|x| (&x).try_into().ok())
+                    .flatten()
```
Comment (birchmd) [inline src/fungible_token.rs:251]:
```suggestion
                    .and_then(|x| (&x).try_into().ok())
```

**Review by birchmd — APPROVED** [review]
As @mfornet has already mentioned, explicitly putting quotes around outputs in some places is quite error prone.

Approving for the sake of quickly addressing this bug, but we should definitely find a more robust way to handle this in the future (e.g. functions that are supposed to return json return a `JsonValue` and then output can be always be serialized using the same logic).

**Comment by mfornet** [comment]
Thanks for fixing tests @sept-en

**Pull request merged**